src/app/donation-start/donation-start.component.ts

```diff
@@ -750,6 +757,19 @@ export class DonationStartComponent implements AfterContentChecked, AfterContent
     }
   }
 
+  onBillingPostCodeChanged(event: Event) {
+    if (this.stripeResponseErrorCode === 'incorrect_zip') {
+      this.stripeError = undefined;
+      this.stripeResponseErrorCode = undefined;
+    }
+
+    if (this.donation) {
+      // Update the donation billingPostalAddress to ensure re-attempt of Stripe payment has the new post code
+      // See ticket DON-493 for more information.
+      this.donation.billingPostalAddress = (<HTMLInputElement>event.target).value;
```

Comment:
This is a bit odd @alihejazi.

Firstly we already have this data bound to nice Angular-known form group properties as used throughout this file – we shouldn't have to resort to a new way of reading it that bypasses Angular Forms.

Secondly, why are we having to do this again just for this field? I think we should go back and work out why `submit()` is not behaving here when it seemingly works for all other data. In particular, I suggest having a look at the condition `... !this.donation?.billingPostalAddress && ...`

This part of the checks was first added on https://github.com/thebiggive/donate-frontend/commit/9db9dbc7a6a4ebf65075d72ab8492c56560259c7 but looks overzealous now – it means we don't react to new billing info at all if we had a billing postal address already. This is tied to avoiding regression in Apple Pay so we should be careful and think about whether in PRB use cases we need a new check to replace this one, but I think it's safe to say we should rationalise the current one.

We then won't need a separate, additional way that we patch `this.donation`. We _might_ not even need the `(change)` event – we can maybe just clear `stripeResponseErrorCode` at the start of `submit()`?